### DISCUSSION TITLE
How to avoid null reference warnings

### DISCUSSION THREAD

**Discussion by lucas-garrido** [linq2db/linq2db#3909] [Q&A] [answered]
Hello, 
i recently started using the nullable reference types.
is there a way to avoid these warnings ?
thanks 

![image](https://user-images.githubusercontent.com/48165310/212112900-db43b4ae-c942-477f-8eff-cd65b93fc696.png)

![image](https://user-images.githubusercontent.com/48165310/212113592-69962ec3-32f0-4e56-851c-9a93f6e5f036.png)

**Comment by MaceWindu** [comment, marked as answer]
For now - only using `!` supporession operator. We should update NRT annotations for those API, I will create tracking issue

**Reply by MaceWindu** [reply]
As for second screenshot - you should use `!` operator. Expressions unfortunatelly doesn't support `?.` operator

**Reply by MaceWindu** [reply]
Another option if you don't use association properties outside queries - you can define them as non-nullable. This definitely makes sense for collection-based associations
```cs
public Address[] Addresses { get; set; } = null!;
// or even better
public IEnumerable<Address> Addresses => throw new InvalidOperationException("Query-only property");
```

for single-record associations I personally prefer to keep NRT annotations to see wether it's inner or outer join association

**Reply by lucas-garrido** [reply]
thanks for your help 🙏im going to do this
